test/system/001-basic.bats

```diff
@@ -280,4 +280,17 @@ run_podman --noout system connection ls
     is "${lines[-1]}" ".*Shutting down engines"
 }
 
+@test "podman - BoltDB cannot create new databases" {
+    OLDDB=$CI_DESIRED_DATABASE
```

Comment:
@edsantiago I'm pretty sure I'm doing this wrong

Reply:
Hoo boy. Coupla things.

First: `$CI_DESIRED_DATABASE` does not tell _podman_ to use boltdb/sqlite, it tell *our Cirrus setup* to do magic setup. So this test is a NOP.

Second, and much more importantly, the way to test this would be `--db-backend` and please allow me to remind you of #20563. If any test ever does `--db-backend=boltdb`, podman is forever hosed, not even `system reset` will fix things. 

This is going to be really hard to test. My recommendation:
1. Revert your `.cirrus.yml` changes 
2. Keep your Go changes, but make the bypass be `$CI_DESIRED_DATABASE`. Stick with what works.
3. Revert the system test. I think this has to be done in an e2e test. If it has to be done in system tests, it might be necessary to use `--runroot --root --this --that --make --sure --you --use --every --tmpdir --available`

But then again I might be missing something (See above).

Reply:
Test suggestion:
```bash
@test "podman - BoltDB cannot create new databases" {
    safe_opts=$(podman_isolation_opts ${PODMAN_TMPDIR})

    CI_DESIRED_DATABASE= run_podman 125 $safe_opts --db-backend=boltdb info
    assert "$output" =~ "deprecated, no new BoltDB databases can be created" \
           "without CI_DESIRED_DATABASE"

    CI_DESIRED_DATABASE=boltdb run_podman $safe_opts --db-backend=boltdb info
    assert "$output" =~ "deprecated; proceeding due to" \
           "with CI_DESIRED_DATABASE"
}
```
You have to add this to `005-info.bats` (or elsewhere), not `001`, because `001` is super paranoid safe and so it does not load `helpers.bash` and so it does not get `podman_isolation_opts`.

This of course assumes that you accept my suggestions above. Modify as needed.

Reply:
@edsantiago Thanks bunches. Added a `system reset` after to try and purge the remains of the tmpdir Podman, which I think ought to be safe this early in the test suite